src/js/account/components/API/__tests__/APIKeys.test.tsx

```diff
@@ -70,7 +73,7 @@ describe("<APIKeys />", () => {
 
         it("should render correctly when newKey is set", async () => {
             const scope = mockApiCreateAPIKey("test", createFakePermissions({ remove_job: true }));
-            renderWithMemoryRouter(<APIKeys />, [{ state: { createAPIKey: true } }]);
+            renderWithMemoryRouter(<APIKeys />, "?openCreateKey=true");
```

Comment:
Fair point on the naming. Prior to this PR browser router was being used for some tests but that was fixed.

re the other points: Another option here could be a RR style abstraction that expects an object for search params and path rather than a string literal. I worry about it being a bit clunky though: couple options:

1. 
```javascript
 renderWithMemoryRouter(ui: ReactNode, path: string, searchParams: object)

```
called like
```javascript
renderWithMemoryRouter(<APIKeys />, "/", {CreateAPIKey: false});
```
or

2.

```javascript
 renderWithMemoryRouter(ui: ReactNode, location: {path: string, searchParams: Object})

```
called like
```javascript
renderWithMemoryRouter(<APIKeys />, {path: "/", searchParams: {CreateAPIKey: false}});

```
In either case the memoryRouter component will convert the passed parameters into at URL. imo version 2 is more verbose than it needs to be given that we can only show path and search params in the URL